src/sentry/workflow_engine/models/incident_groupopenperiod.py

```diff
@@ -16,8 +19,19 @@ class IncidentGroupOpenPeriod(DefaultFieldsModel):
     __relocation_scope__ = RelocationScope.Excluded
 
     incident_id = BoundedBigIntegerField(null=True, unique=True)
+    incident_identifier = models.IntegerField(null=True)
     group_open_period = FlexibleForeignKey("sentry.GroupOpenPeriod", unique=True)
 
     class Meta:
         db_table = "workflow_engine_incidentgroupopenperiod"
         app_label = "workflow_engine"
+        constraints = [
+            models.CheckConstraint(
+                condition=Q(incident_identifier__isnull=False) & Q(incident_id__isnull=False),
+                name="inc_id_inc_identifier_together",
+            ),
```

Comment:
This is saying that `incident_identifier is not null and incident_id is not null`, so it will break any existing rows where these are null. 

I think you want `(incident_identifier is not null and incident_id is not null) or (incident_identifier is null and incident_id is null)`